spec/models/reservation_spec.rb

```diff
@@ -1,725 +1,913 @@
-      FactoryGirl.build(:valid_reservation, reserver_id: banned.id)
+  describe '#flag' do
+    let!(:res) { FactoryGirl.build_stubbed(:valid_reservation) }
+    it 'flags the reservation' do
+      expect { res.flag(:request) }.to \
+        change { res.flagged?(:request) }.from(false).to(true)
     end
-
-    it { is_expected.not_to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_falsey
+    it 'does nothing if flag is undefined' do
+      expect { res.flag(:garbage) }.not_to change { res.flags }
     end
-    it 'fails appropriate validations' do
-      expect(reservation.validate).not_to eq([])
+    it 'does nothing if flag is already set' do
+      res.flag(:request)
+      expect { res.flag(:request) }.not_to change { res.flags }
     end
-    it 'passes other custom validations' do
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.matched_item_and_model).to be_nil
-      expect(reservation.available).to be_nil
-      expect(reservation.not_in_past).to be_nil
+    it "doesn't save the reservation" do
+      allow(res).to receive(:save)
+      allow(res).to receive(:save!)
+      res.flag(:request)
+      expect(res).not_to have_received(:save)
+      expect(res).not_to have_received(:save!)
     end
   end
 
-  context 'with equipment item available problems' do
-    before(:each) do
-      FactoryGirl.create(:checked_out_reservation,
-                         equipment_model: reservation.equipment_model,
-                         start_date: reservation.start_date,
-                         due_date: reservation.due_date,
-                         equipment_item: reservation.equipment_model
-                           .equipment_items.first)
+  describe '#unflag' do
+    let!(:res) { FactoryGirl.build_stubbed(:valid_reservation) }
+    it 'unflags the reservation' do
+      res.flag(:request)
+      expect { res.unflag(:request) }.to \
+        change { res.flagged?(:request) }.from(true).to(false)
     end
-
-    it { is_expected.not_to be_valid }
-    it 'should not save' do
-      expect(reservation.save).to be_falsey
-      expect(Reservation.all.size).to eq(1)
-    end
-    it 'cannot be updated' do
-      reservation.start_date = Time.zone.today + 1.day
-      expect(reservation.save).to be_falsey
+    it 'does nothing if flag is undefined' do
+      expect { res.unflag(:garbage) }.not_to change { res.flags }
     end
-    it 'fails appropriate validations' do
-      expect(reservation.available).not_to eq([])
-      expect(reservation).not_to be_valid
+    it 'does nothing if not flagged' do
+      expect { res.unflag(:request) }.not_to change { res.flags }
     end
-
-    it 'passes other custom validations' do
-      expect(reservation.start_date_before_due_date).to be_nil
-      expect(reservation.not_empty).to be_nil
-      expect(reservation.not_in_past).to be_nil
+    it "doesn't save the reservation" do
```

Comment:
I'll remove these tests, that makes sense.

The only method that saves the reservation seems to be `renew`